nsxt/resource_nsxt_failure_domain.go

```diff
@@ -0,0 +1,156 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/model"
+)
+
+var edgeServices = []string{
+	"active",
+	"standby",
+	"no_preference",
+}
+
+func resourceNsxtFailureDomain() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtFailureDomainCreate,
+		Read:   resourceNsxtFailureDomainRead,
+		Update: resourceNsxtFailureDomainUpdate,
+		Delete: resourceNsxtFailureDomainDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"display_name": getDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"preferred_edge_services": {
+				Type:         schema.TypeString,
+				Description:  "Set preference for failure domain",
+				Optional:     true,
+				Default:      "no_preference",
+				ValidateFunc: validation.StringInSlice(edgeServices, false),
+			},
+		},
+	}
+}
+
+func resourceNsxtFailureDomainRead(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("Error obtaining FailureDomain ID")
+	}
+
+	client := nsx.NewFailureDomainsClient(connector)
+	obj, err := client.Get(id)
+	if err != nil {
+		return fmt.Errorf("error during Failure Domain read: %v", err)
```

Comment:
Some errors are capitalized while others aren't - in the provider project there's no standard for this but let's be consistent at least at the PR level (I'd rather non-capitalized but as you want).

Reply:
Sounds good. Done.